src/main/windows.ts

```diff
@@ -142,6 +147,14 @@ export const openAppWindow = (app: LaunchableApp) => {
             electronApp.quit();
         }
     });
+
+    // @ts-expect-error Custom event
+    appWindow.once('restart-window', () => {
```

Comment:
Is it on purpose, that this uses `once()` while in above on line 60 `on()` is used?

Reply:
Yes, the launcher renderer window will not restart, but reload, it doesn't use any node bindings that are potential WSOD traps.

Reply:
So, above we _must_ use `on()` because it can be invoked multiple times. Here we can use `once()` because a new handler will be registered on the new window (`on()` would also be fine here, because the old window is closed anyway).

`once()` often seems like a strange optimisation to me and often makes me puzzle: “Is this valid? Will it really only be invoked once?”. It is valid here, of course, it just confused me.

Reply:
I used `once()` in case there is a slight possibility that a user double presses Ctrl+R before the window reopens (very unlikely). In which case it would open two new windows.